airbyte-cdk/python/airbyte_cdk/sources/connector_state_manager.py

```diff
@@ -14,24 +27,27 @@ class ConnectorStateManager:
     interface. It also provides methods to extract and update state
```

Comment:
You are right that we only emit stream state from the connector and therefore should only receive per-stream back, so it isn't strictly necessary.

But the changes needed to support building `per-stream` and `shared_state` from a GLOBAL message is a pretty small lift during initializationand doesn't conflict with per-stream. I think it's worth accommodating it because it is part of the Airbyte Protocol that's documented.